core/objects.go

```diff
@@ -49,6 +51,40 @@ const (
 	IdentifierDNS = IdentifierType("dns")
 )
 
+func cmpStrSlice(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	sort.Strings(a)
+	sort.Strings(b)
+	for i := range a {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+	return true
+}
+
+func cmpExtKeyUsageSlice(a, b []x509.ExtKeyUsage) bool {
+	if len(a) != len(b) {
+		return false
+	}
+	intA := make([]int, len(a))
+	intB := make([]int, len(b))
+	for i := range a {
+		intA[i] = int(a[i])
+		intB[i] = int(b[i])
+	}
+	sort.Ints(intA)
+	sort.Ints(intB)
+	for i := range a {
+		if a[i] != b[i] {
+			return false
+		}
+	}
+	return true
+}
```

Comment:
It might be slightly more efficient to use the "throw `a` in a `map[int]bool` and test `b` against it" pattern here, but I'll let this go :)